bundle/jsky.app.ot/src/main/java/jsky/app/ot/viewer/SPElevationPlotPlugin.java

```diff
@@ -234,98 +219,103 @@ private void _setColorCode(String label) {
-            }
-            IDBDatabaseService db = SPDB.get();
-            for (int i = 0; i < _selectedObservations.length; i++) {
-                SPNodeKey progKey = _selectedObservations[i].getProgramKey();
-                ISPProgram prog = db.lookupProgram(progKey);
-                // LORD OF DESTRUCTION: DataObjectManager get without set
-                SPProgram spProg = (SPProgram) prog.getDataObject();
-                queueBand = spProg.getQueueBand();
-                if (queueBand == null || queueBand.length() == 0)
-                    queueBand = "Default Band";
-                else
-                    queueBand = "Band " + queueBand;
-                if ((colors[i] = (Paint) paintMap.get(queueBand)) == null) {
+                for (int i = 1; i < 5; i++) {
                     color = _COLORS[colorIndex++];
+                    queueBand = "Band " + i;
                     paintMap.put(queueBand, color);
-                    colors[i] = color;
                     lic.add(new LegendItem(queueBand, color));
                 }
-            }
-        } else if (_colorCodeTrajectories.equals(_COLOR_CODE_PRIORITY)) {
-            // get the order of items right
-            Paint color;
-            for (SPObservation.Priority pr: _PRIORITIES) {
-                color = _COLORS[colorIndex++];
-                paintMap.put(pr, color);
-                String label = pr.displayValue() + " Priority";
-                lic.add(new LegendItem(label, color));
-            }
-            for (int i = 0; i < _selectedObservations.length; i++) {
-                // LORD OF DESTRUCTION: DataObjectManager get without set
-                final SPObservation spObs = (SPObservation) _selectedObservations[i].getDataObject();
-                final SPObservation.Priority prio = spObs.getPriority();
-                if ((colors[i] = (Paint) paintMap.get(prio)) == null) {
-                    color = _COLORS[colorIndex++];
-                    paintMap.put(prio, color);
-                    colors[i] = color;
-                    String label = prio.displayValue() + " Priority";
-                    lic.add(new LegendItem(label, color));
+                IDBDatabaseService db = SPDB.get();
+                for (int i = 0; i < _selectedObservations.length; i++) {
+                    SPNodeKey progKey = _selectedObservations[i].getProgramKey();
+                    ISPProgram prog = db.lookupProgram(progKey);
+                    // LORD OF DESTRUCTION: DataObjectManager get without set
+                    SPProgram spProg = (SPProgram) prog.getDataObject();
+                    queueBand = spProg.getQueueBand();
+                    if (queueBand == null || queueBand.length() == 0)
+                        queueBand = "Default Band";
+                    else
+                        queueBand = "Band " + queueBand;
+                    if ((colors[i] = paintMap.get(queueBand)) == null) {
+                        color = _COLORS[colorIndex++];
```

Comment:
I think this needs to be `++colorIndex % _COLORS.length` here and elsewhere. It seems to be wrong in every usage. The array must be big enough that we have never hit the end of it.

Reply:
This is worth looking into. I'll check it.

Reply:
Maybe print out the length of `_COLORS` and see what we're dealing with. My guess is 256 ;-)

Reply:
I'm surprised this hasn't caused an `ArrayIndexOutOfBoundsException` that has been noted before.